src/sinks/util/tcp.rs

```diff
@@ -291,6 +291,9 @@ where
             let _open_token = OpenGauge::new().open(|count| emit!(ConnectionOpen { count }));
 
             let mut mapped_input = stream::once(ready(item)).chain(&mut input).map(|event| {
+                drop(event.finalizers);
+                self.acker.ack(1);
+
```

Comment:
I think this pattern (not your addition, but the existing setup) is questionable. If I'm reading right, this effectively does `sink.send_all(once(ready(item)))`, which means to send a stream containing a single item, followed by the stuff in the `chain` closure. I think that should be the same as just `sink.send(item)` followed by the rest of it, which makes for much simpler logic. Does that make sense?

Reply:
Maybe? Are you referring to just ln293 or the literal `sink.send_all` that happens down at ln311?

Reply:
The whole thing, lines 293 through 314.

Reply:
028ce2a3a92335e9c36fe29f3aa3344614b70aee

Reply:
Coming back to this after some other work - I'm not sure I grok the change. Do you have time today to pair on it really quickly?

Reply:
Sure.

Reply:
Recapping here - we misunderstood what was happening in the code here, and it should be fine to move forward